File: UniShare.API/Models/DTOs/Deposits/DepositDto.cs
namespace UniShare.API.Models.DTOs.Deposits;

public class DepositDto
{
    public Guid Id { get; set; }
    public Guid RentalRequestId { get; set; }
    public decimal Amount { get; set; }
    public string Status { get; set; } = null!;
    public string? PaymentProvider { get; set; }
    public string? ProviderTransactionId { get; set; }
    public DateTime? PaidAt { get; set; }
    public DateTime? RefundedAt { get; set; }
    public DateTime CreatedAt { get; set; }
}

File: UniShare.APP/lib/features/deposits/models/deposit_dto.dart
import 'package:json_annotation/json_annotation.dart';
import '../../../core/enums/deposit_status.dart';

part 'deposit_dto.g.dart';

@JsonSerializable()
class DepositDto {
  final String id;
  final String rentalRequestId;
  final double amount;
  final DepositStatus status;
  final String? paymentProvider;
  final String? providerTransactionId;
  final DateTime? paidAt;
  final DateTime? refundedAt;

  const DepositDto({
    required this.id,
    required this.rentalRequestId,
    required this.amount,
    required this.status,
    this.paymentProvider,
    this.providerTransactionId,
    this.paidAt,
    this.refundedAt,
  });

  factory DepositDto.fromJson(Map<String, dynamic> json) =>
      _$DepositDtoFromJson(json);

  Map<String, dynamic> toJson() => _$DepositDtoToJson(this);
}
